### ISSUE TITLE
Offset tooltip and drag/drop position for touch based input

### ISSUE THREAD

**Issue by catlowlevel** [ocornut/imgui#8036]
### Version/Branch of Dear ImGui:

docking

### Back-ends:

imgui_impl_android.cpp + imgui_impl_opengl3.cpp

### Compiler, OS:

ndk, Android

### Details:

because the content of the tooltip is kind of hard to see if the input is touch based (blocked by finger), i would like to offset the tooltip position.
right now im using this code
```cpp
void ImGuiEx::HelpMarker(const char *desc)
{
    ImGui::TextDisabled("(?)");
    ImGuiIO &io = ImGui::GetIO();
    // offset the tooltip 
    if (ImGui::IsItemHovered(ImGuiHoveredFlags_ForTooltip) && io.MouseSource == ImGuiMouseSource_TouchScreen)
    {
        ImVec2 pos = io.MousePos;
        pos.y -= ImGui::GetFontSize() * 2;
        ImGui::SetNextWindowPos(pos, ImGuiCond_Always, ImVec2(0.5f, 1.0f));
    }
    if (ImGui::BeginItemTooltip())
    {
        ImGui::PushTextWrapPos(ImGui::GetFontSize() * 35.0f);
        ImGui::TextUnformatted(desc);
        ImGui::PopTextWrapPos();
        ImGui::EndTooltip();
    }
}
```

is this the correct way? im also using this with `BeginDragDropSource`, however, the drop target location remain in the touch position, which again is kind of blocked by the finger, so is it possible to also offset the `BeginDragDropTarget` accept position?

**Comment by ocornut** [comment]
> is this the correct way? 

it is not incorrect, but I think the issue you are pointing at would be best corrected in the library itself and by default, so we should investigate that.

> im also using this with BeginDragDropSource, however, the drop target location remain in the touch position

You could technically call `SetNextWindowPos()` before `BeginDragDropSource()`. But ditto I think it should be done in the library.

**Comment by ocornut** [comment]
It's not as simple as expected for a surprising reason: using a position pivot that is not (0,0) make it visible and problematic when a tooltip change from one another, e.g. color preview tooltip vs color drag tooltip. I'll need more time to fix it nicely.

**Comment by ocornut** [comment]
This broke positioning of tooltip when using keyboard/gamepad navigation.

**Comment by ocornut** [comment]
Fixed with 50de550
